plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -1435,6 +1436,26 @@ public OptionalInt getMaxWriteParallelism(ConnectorSession session)
         return OptionalInt.of(getWriteParallelism(session));
     }
 
+    protected OptionalInt getMaxColumnNameLengthFromDatabaseMetaData(ConnectorSession session)
```

Comment:
Sorry for 'late notice' and sending this after all the other comments. It only occurred to me to ask this question after I detached myself from the keyboard.

I presume you are not enabling this for all the databases due to 'security reasons', in the sense you want to limit the impact of this change in case something goes wrong?

Is the target state then to have the same call to this method in every Client?

It really feels like what we should do here is detach the code deployment from the rollout (for instance via a feature toggle).

Reply:
here's the process I followed:

- add a test which exposes the problem
- add a mechanism to customise the lengths for each database since the tests showed there's no single value for all of them
- re-run the test with the default lenght being unlimited
- override this method with implementation only in databases where there was a failure

at this point we know (via tests) that the ones without the override do actually support identifiers longer than 65k characters.
we also know that by enabling this path by default for all connectors would incur I/O even in connectors where there's no limit. i.e. we'll ask the DB what the limit is - it'll say no limit. We already know this to be true statically so why incur the I/O and slow down query planning and analysis.